spring-cloud-dataflow-admin/src/main/java/org/springframework/cloud/dataflow/admin/controller/StreamController.java

```diff
@@ -286,37 +284,19 @@ private ModuleDefinition postProcessLibraryProperties(ModuleDefinition module) {
 				.build();
 	}
 
-	private ArtifactType determineModuleType(ModuleDefinition moduleDefinition, boolean isFirst, boolean isLast) {
-		ArtifactType type = null;
-		if (isLast) {
-			if (moduleDefinition.getParameters().containsKey(BindingProperties.OUTPUT_BINDING_KEY)) {
-				if (isFirst) {
-					// single module stream with a named channel in the sink position
-					type = ArtifactType.source;
-				}
-				else {
-					type = ArtifactType.processor;
-				}
-			}
-			else {
-				// this will handle any case that the module is a sink,
-				// even if a single module stream with a named input channel
-				type = ArtifactType.sink;
-			}
-		}
-		else if (isFirst) {
-			if (moduleDefinition.getParameters().containsKey(BindingProperties.INPUT_BINDING_KEY)) {
-				// stream begins with a named channel in the source position
-				type = ArtifactType.processor;
-			}
-			else { 
-				type = ArtifactType.source;
-			}
-		}
-		else {
-			type = ArtifactType.processor;
+	private ArtifactType determineModuleType(ModuleDefinition moduleDefinition) {
```

Comment:
The "module" types (source, processor, and sink) are still relevant as a logical type within a stream, and they are a subset of artifact types (which currently includes "task" and "library" as well). I agree with Eric that this method should be clear that it only returns module types.